lib/chef/provider/package/freebsd/port.rb

```diff
@@ -26,32 +26,44 @@ class Port < Base
           include PortsHelper
 
           def install_package(name, version)
-            shell_out_with_timeout!("make -DBATCH install clean", :timeout => 1800, :env => nil, :cwd => port_dir).status
+            [name].flatten.each do |name|
+              shell_out_with_timeout!("make -DBATCH install clean", :timeout => 1800, :env => nil, :cwd => port_dir(name)).status
+            end
           end
 
           def remove_package(name, version)
-            shell_out_with_timeout!("make deinstall", :timeout => 300, :env => nil, :cwd => port_dir).status
+            [name].flatten.each do |name|
+              shell_out_with_timeout!("make deinstall", :timeout => 300, :env => nil, :cwd => port_dir(name)).status
+            end
           end
 
-          def current_installed_version
-            pkg_info = if @new_resource.supports_pkgng?
-                         shell_out_with_timeout!("pkg info \"#{@new_resource.package_name}\"", :env => nil, :returns => [0, 70])
-                       else
-                         shell_out_with_timeout!("pkg_info -E \"#{@new_resource.package_name}*\"", :env => nil, :returns => [0, 1])
-                       end
-            pkg_info.stdout[/^#{Regexp.escape(@new_resource.package_name)}-(.+)/, 1]
+          # The name of the package (without the version number) as understood by pkg_add and pkg_info.
+          def package_name
+            [@new_resource.package_name].flatten.map do |name|
```

Comment:
BTW, I was looking at apt_package implementation and I wrote the similar way. I Did not know it is wrong. Anyway, without arrays it is much better.